### PR TITLE
Client and Server Status Update

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: 4985_CommAudio/client.cpp [hunk 1]
```diff
@@ -61,20 +63,28 @@ void runTCPClient(ClientWindow *cw, const char *ip, int port)
...
+    }
+
 
     // Initialize address info
     memset((char *)&addr, 0, sizeof(SOCKADDR_IN));
     addr = clientCreateAddress(ip, port);
 
     // Connect to the server
-    if (!connectToServer(sck, &addr))
+    if (!connectToServer(sck, &addr)){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: Weren't you the one pushing for braces on newline?

File: 4985_CommAudio/client.cpp [hunk 2]
```diff
@@ -61,20 +63,28 @@ void runTCPClient(ClientWindow *cw, const char *ip, int port)
     clientWind = cw;    //Init the global ClientWindow
 
     // Create a TCP socket
-    if ((sck = createSocket(SOCK_STREAM, IPPROTO_TCP)) == NULL)
+    if ((sck = createSocket(SOCK_STREAM, IPPROTO_TCP)) == NULL){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/client.cpp [hunk 3]
```diff
@@ -61,20 +63,28 @@ void runTCPClient(ClientWindow *cw, const char *ip, int port)
...
+        cw->updateClientStatus("Status: Socket Error");
         return;
+    }
 
     // Check for a valid host
-    if (!connectHost(ip))
+    if (!connectHost(ip)){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/client.cpp [hunk 4]
```diff
@@ -132,35 +146,45 @@ void runUDPClient(ClientWindow *cw, const char *ip, int port)
     int flag = 1;               //True flag
 
     // Create a UDP socket
-    if ((sck = createSocket(SOCK_DGRAM, IPPROTO_UDP)) == NULL)
+    if ((sck = createSocket(SOCK_DGRAM, IPPROTO_UDP)) == NULL){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/client.cpp [hunk 5]
```diff
@@ -132,35 +146,45 @@ void runUDPClient(ClientWindow *cw, const char *ip, int port)
...
+        cw->updateClientStatus("Status: Socket Error");
         return;
+    }
 
     // Check for valid host
-    if (!connectHost(ip))
+    if (!connectHost(ip)){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/client.cpp [hunk 6]
```diff
@@ -132,35 +146,45 @@ void runUDPClient(ClientWindow *cw, const char *ip, int port)
...
         return;
+    }
 
     // Init address info
     memset((char *)&addr, 0, sizeof(SOCKADDR_IN));
     addr = clientCreateAddress(ip, port);
 
     // Set the reuse addr
-    if (!setCltOptions(sck, SO_REUSEADDR, (char *)&flag))
+    if (!setCltOptions(sck, SO_REUSEADDR, (char *)&flag)){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/client.cpp [hunk 7]
```diff
@@ -132,35 +146,45 @@ void runUDPClient(ClientWindow *cw, const char *ip, int port)
...
+        cw->updateClientStatus("Status: Socket Error");
         return;
+    }
 
     memset((char *)&srvAddr, 0, sizeof(SOCKADDR_IN));
     srvAddr = serverCreateAddress(MCAST_PORT);
 
     // Bind to multicast group
-    if (!bindSocket(sck, &srvAddr))
+    if (!bindSocket(sck, &srvAddr)){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/client.cpp [hunk 8]
```diff
@@ -132,35 +146,45 @@ void runUDPClient(ClientWindow *cw, const char *ip, int port)
...
         return;
+    }
 
     // Setup multicast interface
     stMreq.imr_multiaddr.s_addr = inet_addr(MCAST_ADDR);
     stMreq.imr_interface.s_addr = INADDR_ANY;
 
     // Join multicast group
-    if (!setServOptions(sck, IP_ADD_MEMBERSHIP, (char *)&stMreq))
+    if (!setServOptions(sck, IP_ADD_MEMBERSHIP, (char *)&stMreq)){
```
Comment (RobertArendac) [inline 4985_CommAudio/client.cpp]: brace on newline

File: 4985_CommAudio/clientwindow.h [hunk 9]
```diff
@@ -1,38 +1,45 @@
...
-
-private:
-    Ui::ClientWindow *ui;
-};
-
-#endif // CLIENTWINDOW_H
+        /* clientwindow.h */
+
+    #ifndef CLIENTWINDOW_H
+    #define CLIENTWINDOW_H
```
Comment (RobertArendac) [inline 4985_CommAudio/clientwindow.h]: Why is everything in this file indented one extra??

File: 4985_CommAudio/server.cpp [hunk 10]
```diff
@@ -54,33 +56,40 @@ void runTCPServer(ServerWindow *sw, int port)
     SOCKADDR_IN addr, clientAddr;
 
     // Create socket for listening
-    if ((listenSocket = createSocket(SOCK_STREAM, IPPROTO_TCP)) == NULL)
+    if ((listenSocket = createSocket(SOCK_STREAM, IPPROTO_TCP)) == NULL){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 11]
```diff
@@ -54,33 +56,40 @@ void runTCPServer(ServerWindow *sw, int port)
...
+        sw->updateServerStatus("Status: Socket Error");
         return;
+    }
 
     // Initialize address info
     addr = serverCreateAddress(port);
 
     // Bind the listening socket
-    if (!bindSocket(listenSocket, &addr))
+    if (!bindSocket(listenSocket, &addr)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 12]
```diff
@@ -54,33 +56,40 @@ void runTCPServer(ServerWindow *sw, int port)
...
+        sw->updateServerStatus("Status: Socket Error");
         return;
+    }
 
     // Set socket to listen for connection
-    if (!listenConnection(listenSocket))
+    if (!listenConnection(listenSocket)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 13]
```diff
@@ -54,33 +56,40 @@ void runTCPServer(ServerWindow *sw, int port)
...
+        sw->updateServerStatus("Status: Socket Error");
         return;
+    }
 
+    sw->updateServerStatus("Status: ON");
     // Accept incoming connections and put each client on thread
     while (1)
     {
-        if (!acceptingSocket(&acceptSocket, listenSocket, (SOCKADDR *)&clientAddr))
+        if (!acceptingSocket(&acceptSocket, listenSocket, (SOCKADDR *)&clientAddr)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 14]
```diff
@@ -197,27 +207,39 @@ void runUDPServer(ServerWindow *sw, int port)
     addr = serverCreateAddress(port);
 
     // Create a socket for incomming data
-    if ((acceptSocket = createSocket(SOCK_DGRAM, IPPROTO_UDP)) == NULL)
+    if ((acceptSocket = createSocket(SOCK_DGRAM, IPPROTO_UDP)) == NULL){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 15]
```diff
@@ -197,27 +207,39 @@ void runUDPServer(ServerWindow *sw, int port)
...
+        sw->updateServerStatus("Status: Socket Error");
         return;
+    }
 
     // bind the socket
-    if (!bindSocket(acceptSocket, &addr))
+    if (!bindSocket(acceptSocket, &addr)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 16]
```diff
@@ -197,27 +207,39 @@ void runUDPServer(ServerWindow *sw, int port)
...
         return;
+    }
 
     // Multicast interface
     stMreq.imr_multiaddr.s_addr = inet_addr(MCAST_ADDR);
     stMreq.imr_interface.s_addr = INADDR_ANY;
 
     // Join multicast group, specify time-to-live, and disable loop
-    if (!setServOptions(acceptSocket, IP_ADD_MEMBERSHIP, (char *)&stMreq))
+    if (!setServOptions(acceptSocket, IP_ADD_MEMBERSHIP, (char *)&stMreq)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 17]
```diff
@@ -197,27 +207,39 @@ void runUDPServer(ServerWindow *sw, int port)
...
+        sw->updateServerStatus("Status: Socket Error");
         return;
-    if (!setServOptions(acceptSocket, IP_MULTICAST_TTL, (char *)&ttl))
+    }
+    if (!setServOptions(acceptSocket, IP_MULTICAST_TTL, (char *)&ttl)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/server.cpp [hunk 18]
```diff
@@ -197,27 +207,39 @@ void runUDPServer(ServerWindow *sw, int port)
...
+        sw->updateServerStatus("Status: Socket Error");
         return;
-    if (!setServOptions(acceptSocket, IP_MULTICAST_LOOP, (char *)&flag))
+    }
+    if (!setServOptions(acceptSocket, IP_MULTICAST_LOOP, (char *)&flag)){
```
Comment (RobertArendac) [inline 4985_CommAudio/server.cpp]: brace on newline

File: 4985_CommAudio/serverwindow.cpp [hunk 19]
```diff
@@ -219,3 +218,24 @@ void ServerWindow::on_srvSelectPlaylistButton_clicked()
...
+--
+--  DATE:          March 30, 2017
+--
+--  DESIGNER:      Alex Zielinski
+--
+--  PROGRAMMER:    Alex Zielinski
+--
+--  NOTES:
+--      Updates the status label of the client based on the return value returned by
+--      the functions that create the TCP and UDP sockets
```
Comment (RobertArendac) [inline 4985_CommAudio/serverwindow.cpp]: copy/paste error, says client when it should say server

**Review by RobertArendac — CHANGES_REQUESTED** [review]

File: 4985_CommAudio/clientwindow.h (hunk 9 above)
Comment (ghost) [reply 4985_CommAudio/clientwindow.h]: I have no idea why this happened but I changed it.

**Review by RobertArendac — APPROVED** [review]

**Pull request merged**